helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -733,13 +740,23 @@ private RollingUpdateOp rollingUpdateAwaitRunning(final ZooKeeperClient client,
       // then manually undeployed. The job will not get redeployed, so treat this as a failure.
       final Deployment deployment = getDeployment(host, deploymentGroup.getJobId());
       if (deployment == null) {
-        return opFactory.error(
-            "Job unexpectedly undeployed. Perhaps it was manually undeployed?", host);
+        opFactory.failTask();
```

Comment:
Baking this logic into RollingUpdateOpFactory this way seems a bit out of place and conflated, but I see that you need to emit an event AND move to the next task. Looks like it's time for some refactoring of RollingUpdateOpFactory...